### ISSUE TITLE
Custom MySQL user, other than root

### ISSUE THREAD

**Issue by akopper** [haiwen/seafile-docker#197]
Currently it's only possible to connect to a mysql instance as root.

**Comment by 3ndelebu** [comment]
Yes, it would be great to be able to configure an external database server.

**Comment by metawave** [comment]
Agreed, not safe anyway to have the root password of a component in a configuration. Currently, running it with custom /scripts/utils.py mounted, so I could replace the user in https://github.com/haiwen/seafile-docker/blob/d20a40767de91c20adbe576e79ee8586428051a6/scripts_8.0/utils.py#L271

**Comment by vczyh** [comment]
I dont't understand why the root password is required, even if custom user is supported. 
This is not a good design.

**Comment by vnghia** [comment]
FIY, if you inject into the mysql image docker an init script (i.e copy it to `/docker-entrypoint-initdb.d`) like
```
CREATE DATABASE IF NOT EXISTS `ccnet` CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_ci';
CREATE DATABASE IF NOT EXISTS `seahub` CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_ci';

GRANT ALL PRIVILEGES ON `ccnet`.* TO 'seafile'@'%';
GRANT ALL PRIVILEGES ON `seahub`.* TO 'seafile'@'%';

FLUSH PRIVILEGES;
```

You can leave the `INIT_SEAFILE_MYSQL_ROOT_PASSWORD` empty and it will work.

@freeplant do you think we should mention this somewhere in the docker installation document ?

**Comment by vczyh** [comment]
> FIY, if you inject into the mysql image docker an init script (i.e copy it to `/docker-entrypoint-initdb.d`) like
> 
> ```
> CREATE DATABASE IF NOT EXISTS `ccnet` CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_ci';
> CREATE DATABASE IF NOT EXISTS `seahub` CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_ci';
> 
> GRANT ALL PRIVILEGES ON `ccnet`.* TO 'seafile'@'%';
> GRANT ALL PRIVILEGES ON `seahub`.* TO 'seafile'@'%';
> 
> FLUSH PRIVILEGES;
> ```
> 
> You can leave the `INIT_SEAFILE_MYSQL_ROOT_PASSWORD` empty and it will work.
> 
> [@freeplant](https://github.com/freeplant) do you think we should mention this somewhere in the docker installation document ?

https://github.com/haiwen/seafile-docker/blob/67172737003ce160c47c38685036ed18fe51eaca/scripts/scripts_12.0/utils.py#L271

According to my understanding, seafile entrypoint will call  `wait_for_mysql` function. It needs `root` user . What I mean is, can  `seafile` user be used in the funtion?

It might also be that  I misunderstood.

**Comment by vnghia** [comment]
@vczyh I am not sure why but if you think about it, it makes sense eventually. They said that `INIT_SEAFILE_MYSQL_ROOT_PASSWORD` could be removed after the first deployment but the `wait_for_mysql` function runs everytime you start the container. So I think the `wait_for_mysql` function only reports if we could etablish the connection to the server (something like `ping`) without do any authentication against it ?
https://github.com/haiwen/seafile-docker/blob/67172737003ce160c47c38685036ed18fe51eaca/scripts/scripts_12.0/enterpoint.sh#L63
